optuna_dashboard/static/components/GraphContour.tsx

```diff
@@ -0,0 +1,280 @@
+    }
+  }, [study, objectiveId, xAxis, yAxis])
+
+  return (
+    <Grid container direction="row">
+      <Grid item xs={3}>
+        <Grid container direction="column">
+          {study !== null && study.directions.length !== 1 ? (
+            <FormControl component="fieldset" className={classes.formControl}>
+              <FormLabel component="legend">Objective ID:</FormLabel>
+              <Select value={objectiveId} onChange={handleObjectiveChange}>
+                {study.directions.map((d, i) => (
+                  <MenuItem value={i} key={i}>
+                    {i}
+                  </MenuItem>
+                ))}
+              </Select>
+            </FormControl>
+          ) : null}
+          <FormControl component="fieldset" className={classes.formControl}>
+            <InputLabel id="parameter1">X Axis Parameter</InputLabel>
+            <Select value={xAxis} onChange={handleXAxisChange}>
+              {paramnames.map((x) => (
+                <MenuItem value={x} key={x}>
+                  {x}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+          <FormControl component="fieldset" className={classes.formControl}>
+            <InputLabel id="parameter2">Y Axis Parameter</InputLabel>
+            <Select value={yAxis} onChange={handleYAxisChange}>
+              {paramnames.map((x) => (
+                <MenuItem value={x} key={x}>
+                  {x}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+        </Grid>
+      </Grid>
+      <Grid item xs={6}>
+        <div id={plotDomId} />
+      </Grid>
+    </Grid>
+  )
+}
+
+const plotContour = (
+  study: StudyDetail,
+  objectiveId: number,
+  xAxis: string,
+  yAxis: string
+) => {
+  if (document.getElementById(plotDomId) === null) {
+    return
+  }
+
+  const layout: Partial<plotly.Layout> = {
+    title: "Contour",
+    margin: {
+      l: 50,
+      r: 50,
+    },
+  }
+
+  const trials: Trial[] = study !== null ? study.trials : []
+
+  if (trials.length === 0) {
+    plotly.react(plotDomId, [], layout)
+  }
+
+  const filteredTrials = trials.filter(
+    (t) => t.state === "Complete" || t.state === "Pruned"
+  )
+
+  let paramNames = new Set<string>(trials[0].params.map((p) => p.name))
+  filteredTrials.forEach((t) => {
+    paramNames = new Set<string>(
+      t.params.filter((p) => paramNames.has(p.name)).map((p) => p.name)
+    )
+  })
+
+  if (paramNames.size === 0 || paramNames.size === 1) {
+    plotly.react(plotDomId, [])
+    return
+  }
+  if (xAxis === yAxis) {
+    plotly.react(plotDomId, [], {
+      xaxis: {
+        gridcolor: "#f2f5fa",
+        gridwidth: 1,
+        zerolinecolor: "#f2f5fa",
+        zerolinewidth: 1.5,
+      },
+      yaxis: {
+        gridcolor: "#f2f5fa",
+        gridwidth: 1,
+        zerolinecolor: "#f2f5fa",
+        zerolinewidth: 1.5,
+      },
+      plot_bgcolor: "#E5ecf6",
```

Comment:
Could you explain why this `layout` is different with above?

Reply:
Actually, I was just checking if this layout was fine and forgot to assign it to a variable. I'll make the change.